sumologic/data_source_sumologic_http_source.go

```diff
@@ -55,7 +55,7 @@ func dataSourceSumologicHTTPSourceRead(d *schema.ResourceData, meta interface{})
 	c := meta.(*Client)
 
 	id, _ := strconv.Atoi(d.Id())
-	source, err := c.GetSourceName(d.Get("collector_id").(int64), d.Get("name").(string))
+	source, err := c.GetSourceName(int64(d.Get("collector_id").(int)), d.Get("name").(string))
```

Comment:
I'd actually add a check flag `ok` (type conversion second param) and check for both `int` and `int64` in a type switch.

The reason for this is as I understand this, is that this will be an `int` on 64 bit platforms and `int64` on different platforms (e.g. 32bit).

Reply:
@pmalek-sumo Do you intend to do something like this ?


```go
d64, ok64 := d.Get("collector_id").(int64)
var collector_id int64
if ok64 {
  collector_id = d64
} else {
  dint, ok = d.Get("collector_id").(int)
  if !ok {
     return error
  }
  collector_id = int64(dint)
}
```

Reply:
Pretty much but using a `switch` and a type switch:

```go
...
	var collectorId int64
	switch v := d.Get("collector_id").(type) {
	case int:
		collectorId = int64(v)
	case int64:
		collectorId = v
	default:
		return fmt.Errorf("unknown data type of collector_id: %T, value: %v", v, v)
	}

	source, err := c.GetSourceName(collectorId, d.Get("name").(string))
	if err != nil {
		return err
	}
...
```